src/wasm/tree_parse.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#include <fstream>
+#include <nlohmann/json.hpp>
+#include <vector>
+#include <unordered_map>
+#include "tree_parse.hpp"
+
+using json = nlohmann::json;
+
+
+TreeNode parse_json_to_tree(const json& j_node) {
+    TreeNode node;
+    node.name = j_node.value("name", "");
+    node.value = 0;
+    
+    if (j_node.contains("value")) {
+        if (j_node["value"].is_number_unsigned()) {
+            node.value = j_node["value"].get<uint64_t>();
+        } else if (j_node["value"].is_number_integer()) {
+            node.value = static_cast<uint64_t>(j_node["value"].get<int64_t>());
+        }
```

Comment:
Same here: Unless a compiler complains, you can safely assume that ```j_node["value"]``` is an unsigned integer, as a value cannot be a negative number.

Reply:
Done :+1: